source/slang-wasm/slang-wasm.cpp

```diff
@@ -0,0 +1,314 @@
+        if (result != SLANG_OK)
+        {
+            g_error.type = std::string("USER");
+            g_error.result = result;
+            return nullptr;
+        }
+    }
+
+    return new GlobalSession(globalSession);
+}
+
+Session* GlobalSession::createSession()
+{
+
+    ISession* session = nullptr;
+    {
+        SessionDesc sessionDesc = {};
+        sessionDesc.structureSize = sizeof(sessionDesc);
+        constexpr SlangInt targetCount = 1;
+        TargetDesc targets[targetCount] = {
+            {.structureSize = sizeof(TargetDesc), .format = SLANG_WGSL}
+        };
+        sessionDesc.targets = targets;
+        sessionDesc.targetCount = targetCount;
+        SlangResult result = m_interface->createSession(sessionDesc, &session);
+        if (result != SLANG_OK)
+        {
+            g_error.type = std::string("USER");
+            g_error.result = result;
+            return nullptr;
+        }
+    }
+
+    return new Session(session);
+}
+
+Module* Session::loadModuleFromSource(const std::string& slangCode)
+{
+    Slang::ComPtr<IModule> module;
+    {
+        const char * name = "";
+        const char * path = "";
+        Slang::ComPtr<slang::IBlob> diagnosticsBlob;
+        Slang::ComPtr<ISlangBlob> slangCodeBlob = UnownedBlob::create(
+            slangCode.c_str(), slangCode.size());
+        try
+        {
+            module = m_interface->loadModuleFromSource(
+                name, path, slangCodeBlob, diagnosticsBlob.writeRef());
+        }
+        catch(const Slang::AbortCompilationException& e)
+        {
+            // TODO: This currently happens when introducing a simple syntax error
+            //       in the Slang code! Figure out why.
+            g_error.type = std::string("INTERNAL");
+            g_error.message = std::string(e.Message.begin(), e.Message.end());
+            return nullptr;
+        }
+        if (!module) {
+            g_error.type = std::string("USER");
+            g_error.message = std::string(
+                (char*)diagnosticsBlob->getBufferPointer(),
+                (char*)diagnosticsBlob->getBufferPointer() +
+                diagnosticsBlob->getBufferSize());
+            return nullptr;
+        }
+    }
+
+    return new Module(module);
+}
+
+EntryPoint* Module::findEntryPointByName(const std::string& name)
+{
+    Slang::ComPtr<IEntryPoint> entryPoint;
+    {
+        SlangResult result = moduleInterface()->findEntryPointByName(
+            name.c_str(), entryPoint.writeRef());
+        if (result != SLANG_OK)
+        {
+            g_error.type = std::string("USER");
+            g_error.result = result;
+            return nullptr;
+        }
+    }
+
+    return new EntryPoint(entryPoint);
+}
+
+ComponentType* Session::createCompositeComponentType(
+    const std::vector<ComponentType*>& components)
+{
+    Slang::ComPtr<IComponentType> composite;
+    {
+        std::vector<IComponentType*> nativeComponents(components.size());
+        for (
```

Comment:
Style: Everything in a single line

Reply:
Done. (Didn't fit with 90 columns, so I changed the loop index to just `i`.)